internal/common/autogen/resource_base.go

```diff
@@ -0,0 +1,60 @@
+package autogen
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+)
+
+var _ ResourceAPIOperations = &RSAutogen{}
+
+// RSAutogen is used as an embedded struct for all autogen resources.
+type RSAutogen struct {
+	config.RSCommon
+}
+
+// ResourceAPIOperations defines the interface for API operations in autogen resources.
+// RSAutogen provides default implementations which can be overridden by individual resources.
+type ResourceAPIOperations interface {
+	PerformRead(ctx context.Context, req *HandleReadReq) ([]byte, *http.Response, error)
+	PerformCreate(ctx context.Context, req *HandleCreateReq) ([]byte, *http.Response, error)
+	PerformUpdate(ctx context.Context, req *HandleUpdateReq) ([]byte, *http.Response, error)
+	PerformDelete(ctx context.Context, req *HandleDeleteReq) error
```

Comment:
[nit] `Perform` prefix is redundant + golang prefers simple verbs. If you drop the code is still self-explanatory: `...Operations.Delete`